presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+ * **Description:** See :ref:`hive.orc.max-buffer-size <tuning-pref-hive>`.
+
+
+``plan_with_table_node_partitioning``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** **Experimental.** Adapt plan to use backend partitioning. By setting this property you allow to use partitioning provided by table layout itself while collecting required data. This may allow to utilize optimization of table layout provided by specific connector. In particular, when this is set presto will try to partition data for workers in a way that each workers gets a chunk of data that comes from one backend partition. It can be particularly useful due to the I/O distribution optimization in table partitioning. Note that this property may only be utilized if given projection uses all columns used for table partitioning inside connector.
+
+
+``prefer_streaming_operators``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Prefer source table layouts that produce streaming operators. Setting this property will allow workers not to wait for chunks of data to start processing them while scanning tables. This may cause faster processing  with lower latency and downtime but some operators may do things more efficiently when working with chunks of data.
+
+
+``push_table_write_through_union``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``optimizer.push-table-write-through-union`` (``true``)
+ * **Description:** See :ref:`optimizer.push-table-writethrough-union <tuning-pref-optimizer>`.
+
+
+``query_max_memory``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``query.max-memory`` (``20 GB``)
+ * **Description:** This property can be use to be nice to the cluster if a particular query is not as important as the usual cluster routines. Setting this value to less than the server property ``query.max-memory`` will cause presto to drop the query in the session if it will require more then ``query_max_memory`` memory. Setting this value to higher than ``query.max-memory`` will not have any effect.
+
+
+
+``query_max_run_time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``query.max-run-time`` (``100 d``)
+ * **Description:** If the expected query processing time is higher than ``query.max-run-time``, it is crucial to set this session property to prevent results of long running queries being dropped after ``query.max-run-time``. A session may also set this value to lower than ``query.max-run-time`` in order to crosscheck for bugs in query. It may be particularly useful when setting up a session with a very large number of short-running queries. It is important to set this value to much higher than the average query time to avoid problems with outliers (some queries may randomly take much longer due to cluster load and other circumstances).
```

Comment:
s/A session may also set this value to lower than ``query.max-run-time`` in order to crosscheck for bugs in query. It may be particularly useful when setting up a session with a very large number of short-running queries./A session may also set this value to less than ``query.max-run-time`` in order to crosscheck for bugs in the query.  Setting this value less than ``query.max-run-time`` may be particularly useful for a session with a very large number of short-running queries.